### PR TITLE
Today Widget - Tracks

### LINKED ISSUE #15158: Today Widget - Tracks
Add tracks to the Today Widget

### PR DESCRIPTION
**Pull request by Gio2018**
Fixes #15158

To test:
Pre requisites:
- Better to start from scratch and delete any previously installed version if you had tested widgets before.
- Make sure you have iOS 14.3 installed on your device
- In the developer settings, enable "WidgetKit Developer Mode"

1. Build and run this version
2. Install one or more widgets
3. In mission control, select the tracks live view tool (Metrics -> Tracks -> Live view), and input `wpios_today_home_extension_widget_updated` as event name
4. Wait for at least five minutes (sometimes it takes 8/9 minutes) and see the incoming tracks in the log
5. Expand the tracks and verify that the info medium_widgets, small_widgets and total_widgets are consistent with that you installed

Known issue: two widgets of the same size and the same site, seem to be counted as one.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by Gio2018** [comment]
Converting it to `ready for review` as the results look more promising on iOS 14.3. Will still need some testing to confirm it works properly

**Comment by diegoreymendez** [comment]
I'm waiting to review this for now as there are CircleCI issues (and since I'm trying to resolve other unrelated issues).  Let me know @Gio2018 if you still want me to go ahead anyway.

**Comment by Gio2018** [comment]
> I'm waiting to review this for now as there are CircleCI issues (and since I'm trying to resolve other unrelated issues). Let me know @Gio2018 if you still want me to go ahead anyway.

Hey @diegoreymendez , no particular rush on this one. The CircleCI issue is on the installable build that I triggered, and it's likely because it needs to be updated with the latest version of `develop` (not sure why the hound violation is still showing up there as it was already fixed)

File: WordPress/WordPressHomeWidgetToday/Tracks/Tracks+TodayHomeWidget.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        let previousCount = UserDefaults(suiteName: WPAppGroupName)?.object(forKey: WPHomeWidgetTodayCount) as? Int ?? 0
+        let newCount = widgetInfo.count
+        
+        guard previousCount != newCount else {
+            return
+        }
+
+        UserDefaults(suiteName: WPAppGroupName)?.set(newCount, forKey: WPHomeWidgetTodayCount)
+
+        let properties = ["total_widgets": newCount,
```
Comment (frosty) [inline WordPress/WordPressHomeWidgetToday/Tracks/Tracks+TodayHomeWidget.swift:48]: We could maybe add these keys to an enum?

**Review by frosty — APPROVED** [review]
This seems to be working as expected for me, with the exception of the known issue you mentioned 🙂 However, I feel that two identical widgets will be a fairly unlikely use case!

**Pull request merged**